compiler/rustc_target/src/spec/mod.rs

```diff
@@ -211,6 +246,16 @@ impl LinkerFlavor {
     /// of truth, other flags are used in case of ambiguities.
     fn from_cli_json(cli: LinkerFlavorCli, lld_flavor: LldFlavor, is_gnu: bool) -> LinkerFlavor {
         match cli {
+            LinkerFlavorCli::Gnu(cc, lld) => LinkerFlavor::Gnu(cc, lld),
+            LinkerFlavorCli::Darwin(cc, lld) => LinkerFlavor::Darwin(cc, lld),
+            LinkerFlavorCli::WasmLld(cc) => LinkerFlavor::WasmLld(cc),
+            LinkerFlavorCli::Unix(cc) => LinkerFlavor::Unix(cc),
+            LinkerFlavorCli::Msvc(Lld::Yes) => LinkerFlavor::Msvc(Lld::Yes),
```

Comment:
Could you keep the ordering in all `match`es consistent with `enum LinkerFlavorCli`?
I.e. all the non-legacy flavors are above (including msvc) and all legacy flavors are below, no matter stable or not.

(Applies to `linker_flavor_cli_impls!` too.)